cmd/src/actions_exec.go

```diff
@@ -401,14 +401,49 @@ func prepareAction(ctx context.Context, action Action) error {
 // have been pulled from or pushed to a registry. See
 // https://windsock.io/explaining-docker-image-ids/ under "A Final Twist" for a good
 // explanation.
-func getDockerImageContentDigest(ctx context.Context, image string) (string, error) {
+func getDockerImageContentDigest(ctx context.Context, image string, logger *actionLogger) (string, error) {
 	// TODO!(sqs): is image id the right thing to use here? it is NOT the
 	// digest. but the digest is not calculated for all images (unless they are
 	// pulled/pushed from/to a registry), see
 	// https://github.com/moby/moby/issues/32016.
 	out, err := exec.CommandContext(ctx, "docker", "image", "inspect", "--format", "{{.Id}}", "--", image).CombinedOutput()
 	if err != nil {
-		return "", fmt.Errorf("error inspecting docker image (try `docker pull %q` to fix this): %s", image, bytes.TrimSpace(out))
+		// An error could mean the image just didn't exist, so we need to pull it first.
```

Comment:
How do we determine that? What if the error is "docker: command does not exist"? We then log "Pulling image..." which would fail.

Is there are a way to determine whether we should pull the image or return the error?

Reply:
The logic 

- tries to get the digest
- on fail, tries to pull
- tries to get the digest again
- on fail, it's not related to the pull and we return the error

Reply:
Yeah, I know that, but the "on fail" could be caused by anything, right?

If "tries to get the digest" fails because docker is not installed, then we run "docker pull" and print more stuff to stderr.

In other words: can't we check the output returned by "docker image inspect" to see if it contains "No such image" (or something) and only then try to pull?